core/src/test/java/org/apache/calcite/test/LatticeTest.java

```diff
@@ -677,10 +677,10 @@ private void checkTileAlgorithm(String statisticProvider,
             + "join \"foodmart\".\"time_by_day\" using (\"time_id\")\n"
             + "group by \"the_year\"")
         .enableMaterializations(true)
-        .explainContains("EnumerableCalc(expr#0=[{inputs}], expr#1=[IS NOT NULL($t0)], "
-            + "expr#2=[1:BIGINT], expr#3=[0:BIGINT], expr#4=[CASE($t1, $t2, $t3)], C=[$t4])\n"
-            + "  EnumerableAggregate(group=[{0}])\n"
-            + "    EnumerableTableScan(table=[[adhoc, m{32, 36}]])")
+        .explainContains("EnumerableCalc(expr#0..1=[{inputs}], C=[$t1])\n"
+            + "  EnumerableAggregate(group=[{0}], C=[COUNT($0)])\n"
+            + "    EnumerableAggregate(group=[{0}])\n"
```

Comment:
This looks like a plan degradation, doesn't it?

Reply:
It does have the original plan's alternative, but from cost model's perspective, the new one is a cheaper plan.

Reply:
Who will fix the costing model then?
I think it is unfair to merge a change that is not really compatible with the costing model.

If the change to optimizer requires adjustments to the costing model, then could you please do that in a single commit, so we see the net changes for both plans and the response times?

Reply:
I think it is orthogonal. should be done is a separate PR. The issue of cost model exists before this change.